idlparser/src/ffi/types/mod.rs

```diff
@@ -0,0 +1,77 @@
+use crate::types;
+use std::{slice, str};
+
+pub mod visitor;
+
+/// # Safety
+///
+/// See the safity documentation of [`slice::from_raw_parts`].
+#[no_mangle]
+pub unsafe extern "C" fn parse_idl(idl_ptr: *const u8, idl_len: u32) -> *mut Program {
+    let idl = unsafe { slice::from_raw_parts(idl_ptr, idl_len.try_into().unwrap()) };
+    let idl = str::from_utf8(idl).unwrap();
+    let program = types::parse_idl(idl).unwrap();
+    let program = Box::new(program);
+    Box::into_raw(program)
+}
+
+/// # Safety
+///
+/// TODO
+#[no_mangle]
+pub unsafe extern "C" fn free_program(program: *mut Program) {
+    if program.is_null() {
+        return;
+    }
+    unsafe {
+        drop(Box::from_raw(program));
+    }
+}
+
+pub type Program = types::Program;
+
+pub type Service = types::Service;
+
+#[repr(C, packed)]
```

Comment:
Why `packed`? Will it be compatible with FFI?

Reply:
agreed, from the [docs](https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked):

`repr(packed) is not to be used lightly. Unless you have extreme requirements, this should not be used.`

Reply:
Don't we use it a lot in our backend? Like here https://github.com/gear-tech/gear/blob/62cd946ed3a851a4f80deda768df7fa213f5171b/gsys/src/lib.rs#L70
Do you think it is better to use `repr(C)` with native padding?

Reply:
How repr(C) will be achieved in 'JS' @osipov-mit ?

Reply:
> https://github.com/gear-tech/gear/blob/62cd946ed3a851a4f80deda768df7fa213f5171b/gsys/src/lib.rs#L70

I think things are `repr(packed)` in backend because of light memory requirements or some ABI compatibility reasons. We should avoid using `packed` here if removing it doesn't break things.